mtags/src/main/scala-3/scala/meta/internal/pc/CompletionValue.scala

```diff
@@ -2,25 +2,52 @@ package scala.meta.internal.pc
 
 import scala.meta.internal.pc.CompletionValue.Kind
 
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags.*
+import dotty.tools.dotc.core.Symbols.NoSymbol
 import dotty.tools.dotc.core.Symbols.Symbol
 import dotty.tools.dotc.interactive.Completion
+import org.eclipse.lsp4j.CompletionItemKind
 
 case class CompletionValue(
     label: String,
     symbol: Symbol,
-    kind: Kind
-)
+    kind: Kind,
+    isCustom: Boolean = false,
+    insertText: Option[String] = None
```

Comment:
Maybe switch it sealed trait/enum to avoid  dealing with `NoSymbol`?

Reply:
I was thinking about it, but I would rather do it as a separate refactoring, I think this might be even more useful with some additional custom completions. Will do it as a next step.